test/cartesian.jl

```diff
@@ -515,6 +515,11 @@ end
 f39705() = Base.Cartesian.@nany 0 _ -> true
 @test f39705() === false
 
+@testset "Cartesian @nall macro test" begin
+    nall_test = Base.Cartesian.@nall 0 _ -> true
```

Comment:
Could perhaps make this a bit more "difficult" and similar to real world usage, something like:

```julia
i_1, i_2, i_3 = 1, 2, 3;
@test Base.Cartesian.@nall 2 d->(i_d <= 2)
@test !Base.Cartesian.@nall 3 d->(i_d <= 2)